[2025-08-14T20:19:34.611+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [queued]>
[2025-08-14T20:19:34.620+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [queued]>
[2025-08-14T20:19:34.620+0000] {taskinstance.py:2170} INFO - Starting attempt 10 of 10
[2025-08-14T20:19:34.630+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): fetch_historical_data> on 2025-08-14 03:48:52+00:00
[2025-08-14T20:19:34.634+0000] {standard_task_runner.py:60} INFO - Started process 26410 to run task
[2025-08-14T20:19:34.638+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'stock_market_batch_pipeline', 'fetch_historical_data', 'manual__2025-08-14T03:48:52+00:00', '--job-id', '32', '--raw', '--subdir', 'DAGS_FOLDER/stock_market_batch_dag.py', '--cfg-path', '/tmp/tmpc4n90msb']
[2025-08-14T20:19:34.647+0000] {standard_task_runner.py:88} INFO - Job 32: Subtask fetch_historical_data
[2025-08-14T20:19:34.692+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [running]> on host 0181ae2045ff
[2025-08-14T20:19:34.754+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Basanth Kumar' AIRFLOW_CTX_DAG_ID='stock_market_batch_pipeline' AIRFLOW_CTX_TASK_ID='fetch_historical_data' AIRFLOW_CTX_EXECUTION_DATE='2025-08-14T03:48:52+00:00' AIRFLOW_CTX_TRY_NUMBER='10' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-14T03:48:52+00:00'
[2025-08-14T20:19:34.755+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-08-14T20:19:34.757+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n    unset PYTHONNOUSERSITE\n    python /opt/***/dags/scripts/batch_data_producer.py 2025-08-14\n    ']
[2025-08-14T20:19:34.764+0000] {subprocess.py:86} INFO - Output:
[2025-08-14T20:19:35.263+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-08-14T20:19:35.264+0000] {subprocess.py:93} INFO -   File "/opt/***/dags/scripts/batch_data_producer.py", line 8, in <module>
[2025-08-14T20:19:35.264+0000] {subprocess.py:93} INFO -     import yfinance as yf
[2025-08-14T20:19:35.265+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'yfinance'
[2025-08-14T20:19:35.342+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-08-14T20:19:35.352+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-08-14T20:19:35.356+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=stock_market_batch_pipeline, task_id=fetch_historical_data, execution_date=20250814T034852, start_date=20250814T201934, end_date=20250814T201935
[2025-08-14T20:19:35.367+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 32 for task fetch_historical_data (Bash command failed. The command returned a non-zero exit code 1.; 26410)
[2025-08-14T20:19:35.411+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-14T20:19:35.437+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
